src/main/scala/com/gu/hmac/HMACHeaders.scala

```diff
@@ -57,56 +60,73 @@ object HMACDate {
 
 case class HMACHeaderValues(date: String, token: String)
 
-trait HMACHeaders {
+object HMACSignatory {
   import HMACDate.DateTimeOps
-  import HMACToken.TokenOps
 
-  def secret: String
+  final val Algorithm = "HmacSHA256"
+  final val UTF8Charset = StandardCharsets.UTF_8
 
-  private val Algorithm = "HmacSHA256"
-  private val HmacValidDurationInMinutes = 5
-  private val MinuteInMilliseconds = 60000
-  private val UTF8Charset = StandardCharsets.UTF_8
+  def sign(secretKey: String, date: DateTime, uri: URI): String = {
+    val input = List[String](date.toRfc7231String, uri.getPath)
+    val toSign = input.mkString("\n")
+    calculateHMAC(secretKey, toSign)
+  }
 
-  def validateHMACHeaders(dateHeader: String, authorizationHeader: String, uri: URI): Boolean = {
-    val hmacDate = HMACDate.get(dateHeader)
-    val hmacToken = HMACToken.get(authorizationHeader)
-    isDateValid(hmacDate) && isHMACValid(hmacDate, uri, hmacToken)
+  private def calculateHMAC(secretKey: String, toEncode: String): String = {
+    val signingKey = new SecretKeySpec(secretKey.getBytes(UTF8Charset), Algorithm)
+    val mac = Mac.getInstance(Algorithm)
+    mac.init(signingKey)
+    val rawHmac = mac.doFinal(toEncode.getBytes(UTF8Charset))
+    new String(Base64.encodeBase64(rawHmac), UTF8Charset)
   }
+}
 
-  def createHMACHeaderValues(uri: URI): HMACHeaderValues = {
-    val now = DateTime.now()
-    createHMACHeaderValues(uri, now)
+trait CreateHMACHeader {
+  import HMACToken.TokenOps
+  import HMACDate.DateTimeOps
+
+  protected val clock: Clock = Clock.systemDefaultZone()
```

Comment:
Using `Clock` allows consumers the ability to change timezones if required, but also allows us to more easily test these functions by using a fixed Clock.